tests/test_cli.py

```diff
@@ -176,6 +152,6 @@ def test_empty_values_keys_are_always_present(self):
 
 class TestGetBuildTargets(TestCase):
     def test_returns_base_target_and_all_variants(self):
-        mbed_target = mock.Mock(spec_set=MbedTarget, build_variant=("S", "NS"), board_type="FOO")
+        mbed_target = mock.create_autospec(MbedTarget, build_variant=("S", "NS"), board_type="FOO")
```

Comment:
This comment applies to all changes doing `create_autospec`:

1. I'd rather we kept `spec_set`, as it is the actual guard that prevents us from setting attributes that don't exist on the mocked objects. I.e. w/o `spec_set` you can do:
```
mock.create_autospec(MbedTarget, this_really_doesnt_exist_on_mbed_target="foo")
```
And code under test can use non-existent attributes. 😬 

2. I'm not sure what we gain when using `create_autospec` for tests that do simple attribute access. We don't get the argument check for method calls, as there ain't any. Is there something I'm missing here?

Reply:
https://bugs.python.org/issue36580

Reply:
Could you please elaborate?

Even after reading the issue, I don't understand why this change is necessary. The previous way of mocking works as well as the current one. What's the difference here?

Also, your reply doesn't cover `spec_set` so I'm not sure if you ack or not. 😅

Reply:
No the regular mock construction with spec_set won't work as the mock doesn't do signature validation on the `__init__` so they recommend `create_autospec`

Reply:
If you read all the replies you'll see.

Reply:
I think the thing you're missing is dataclasses magically convert the class attributes to instance attributes in the __new__ and __init__ methods and it confuses mock as it doesn't think there are any instance attributes, as they don't appear in `dir` hence aren't copied when using `spec`.

Reply:
It'd probably be smarter if I tried this locally using your branch on `mbed-targets` instead of master. Makes sense, thanks for explanation.

It is somewhat sad (again) that we can't use `spec_set`, as this means you can set arbitrary stuff on the mock, which makes this whole mocking shenanigan even worse. :) I hope this won't bite us in the future.

Reply:
I'd prefer to remove the mocking where possible. I'll look into that today as a "fun" innovation day project